eng/common/scripts/job-matrix/job-matrix-functions.ps1

```diff
@@ -515,15 +519,12 @@ function CreateMatrixCombinationScalar([MatrixParameter[]]$permutation, [Hashtab
 
     # The maximum allowed matrix name length is 100 characters
     $name = $names -join "_"
+    if ($name -and $name[0] -match "^[0-9]") {
```

Comment:
Are there any other special chars in this set?

Reply:
We could match on any non-alpha characters instead of just digits, which would be more flexible.

Reply:
Actually, this is not necessary because the job name will already filter out non alphanumeric characters since they aren't supported at all by devops.